### ISSUE TITLE
[2.1.1] fails to find `./schema/loader.json` (`schema-utils`)

### ISSUE THREAD

**Issue by kimamula** [webpack-contrib/extract-text-webpack-plugin#534]
Updating to 2.1.1 results in the following error.

```js
> Error: ENOENT: no such file or directory, open '/path/to/my/project/schema/loader.json'
>>     at Error (native)
>>     at Object.fs.openSync (fs.js:640:18)
>>     at Object.fs.readFileSync (fs.js:508:33)
>>     at validateOptions (/path/to/my/project/node_modules/schema-utils/dist/validateOptions.js:35:27)
>>     at Function.ExtractTextPlugin.extract (/path/to/my/project/node_modules/extract-text-webpack-plugin/index.js:206:3)
>>     at Object.<anonymous> (/path/to/my/project/webpack.config.js:19:32)
>>     at Module._compile (module.js:570:32)
>>     at Object.Module._extensions..js (module.js:579:10)
>>     at Module.load (module.js:487:32)
>>     at tryModuleLoad (module.js:446:12)
```

It is likely that you should use something like `path.join(__dirname, 'schema/loader.json')` instead of `'./schema/loader.json'` in the following lines.

https://github.com/webpack-contrib/extract-text-webpack-plugin/blob/0271b391490048815087e1325542e80a39eccb09/index.js#L127
https://github.com/webpack-contrib/extract-text-webpack-plugin/blob/0271b391490048815087e1325542e80a39eccb09/index.js#L206

**Comment by michael-ciniawsky** [comment]
@kimamula Could you do me a favour and locally try by omit the `.json`  extname and test if it works ?

**Comment by kimamula** [comment]
No, omitting `.json` does not work.

**Comment by MirrorBytes** [comment]
Using path.resolve rectified the issue for me. @kimamula

**Comment by MirrorBytes** [comment]
Change line 127 to:
`validateOptions(path.resolve(__dirname, './schema/plugin.json'), options, 'Extract Text Plugin');`

Change line 206 to:
`validateOptions(path.resolve(__dirname, './schema/loader.json'), options, 'Extract Text Plugin (Loader)');`

Require path at the top, and everything works fine again.

**Comment by michael-ciniawsky** [comment]
hmm.. i need to check that again :), the paths should be resolved by `validateOptions` https://github.com/webpack-contrib/schema-utils/blob/master/src/validateOptions.js#L15 (but without using `__dirname`)

@MirrorBytes Do want to send a PR with the fix otherwise I do it

**Comment by MirrorBytes** [comment]
The issue is that it's resolving based on the working directory.

**Comment by michael-ciniawsky** [comment]
Until now it worked at least for loaders :), but `__dirname` is seems to be the better solution, anyways we need a quick fix fot this now 😛, please yes/no on sending PR

**Comment by MirrorBytes** [comment]
I'll send it.

**Comment by ahoym** [comment]
Got this error recently, fixed it temporarily by pinning to v2.0.0, though 2.1.0 may also work

**Comment by MirrorBytes** [comment]
Sent it. @michael-ciniawsky

**Comment by dnutels** [comment]
@ahoym 2.1.0 does indeed work.

**Comment by michael-ciniawsky** [comment]
Released in [2.1.2](https://github.com/webpack-contrib/extract-text-webpack-plugin/releases/tag/v2.1.2)
